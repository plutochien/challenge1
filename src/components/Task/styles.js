import styled from 'styled-components';

export const Row = styled.li`
  display: flex;
  width: auto;
  height: 50px;
  align-items: center;
  list-style: none;
  margin: 10px;
  border-bottom: 1px solid #e1e1e1;
  background: white;
  border-radius: 5px;
  padding: 0 0 0 20px;
  font-size: 20px;
`;

export const Text = styled.span`
  flex: 1 1;
  color: #0c5460;
  text-align: justify;
  word-break: break-all;
  text-decoration: ${props =>
    props.isComplete ? 'line-through #cd5c5c' : 'none'};
`;

export const Order = styled.span`
  background: #007bff;
  border-radius: 15px;
  color: #ffffff;
  display: inline-block;
  font-weight: bold;
  line-height: 30px;
  margin-right: 15px;
  text-align: center;
  width: 30px;
  font-weight:200px;
`;

export const Button = styled.button`
  color: white;
  background-color: ${props => (props.isComplete ? '#FFC106' : '#28a745')};
  border-color: ${props => (props.isComplete ? '#FFC106' : '#28a745')};
  text-align: center;
  vertical-align: middle;
  font-weight: 400;
  border: 1px solid transparent;
  padding: 6px 12px;
  font-size: 16px;
  line-height: 1.5;
  border-radius: 4px;
  margin: 5px;
  cursor: pointer;
`;

export const DeleteButton = styled.button`
  color: white;
  background-color: #DC354A;
  border-color: #DC354A;
  text-align: center;
  vertical-align: middle;
  font-weight: 400;
  border: 1px solid transparent;
  padding: 6px 12px;
  font-size: 16px;
  line-height: 1.5;
  border-radius: 4px;
  margin: 5px;
  cursor: pointer;
`;

export const Badge = styled.span`
  color: #0c5460 ;
  background-color: #ffc107;
  border-color: #ffc107;
  display: inline-block;
  padding: 4px 6px;
  font-size: 75%;
  font-weight: 200;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 4px;
  margin-left: 10px;
`;
