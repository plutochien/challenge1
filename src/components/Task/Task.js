import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { deleteTask, toggleComplete } from '../../actions';
import { Badge, Button, DeleteButton, Order, Row, Text } from './styles';

const Task = ({
  data: { id, isComplete, taskText, order, priority },
  deleteTask,
  toggleComplete,
}) => {
  return (
    <div>
      <Row>
        <Order priority={priority}>{order} </Order>
        <Text isComplete={isComplete} priority={priority}>
          {taskText}
          {isComplete ? <Badge>Done</Badge> : ''}
        </Text>

        <Button
          isComplete={isComplete}
          size="40"
          onClick={() => {
            toggleComplete(id);
          }}
        >
          {isComplete ? 'Undone' : 'Done'}
        </Button>
        <DeleteButton
          size="40"
          onClick={() => {
            deleteTask(id);
          }}
        >
          Delete
        </DeleteButton>
      </Row>
    </div>
  );
};

Task.propTypes = {
  data: PropTypes.object.isRequired,
  deleteTask: PropTypes.func.isRequired,
  toggleComplete: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      deleteTask,
      toggleComplete,
    },
    dispatch,
  );

export default connect(null, mapDispatchToProps)(Task);
