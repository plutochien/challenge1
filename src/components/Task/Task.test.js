import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Task from './Task';
const mockStore = configureStore([]);

describe('AddTask Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      todos: {
        list: [
          {
            id: 7,
            taskText: 'task1',
            isComplete: true,
          },
          {
            id: 6,
            taskText: 'task2',
            isComplete: false,
          },
          {
            id: 1,
            taskText: 'task3',
            isComplete: true,
          },
        ],
      },
      filter: 'SHOW_ALL',
    });
    store.dispatch = jest.fn();
    const mockData = {
      id: 1,
      order: 2,
      taskText: 'test task',
      isComplete: false,
    };
    component = renderer.create(
      <Provider store={store}>
        <Task data={mockData} toggleComplete={jest.fn} deleteTask={jest.fn} />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    const mockData = {
      id: 1,
      order: 2,
      taskText: 'test task',
      isComplete: true,
    };
    component = renderer.create(
      <Provider store={store}>
        <Task data={mockData} toggleComplete={jest.fn} deleteTask={jest.fn} />
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on click Done button', () => {
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick();
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });

  it('should dispatch an action on click Delete button', () => {
    renderer.act(() => {
      component.root.findAllByType('button')[1].props.onClick();
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});
