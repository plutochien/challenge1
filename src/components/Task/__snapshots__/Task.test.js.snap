// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask Component should render with given state from Redux store 1`] = `
<div>
  <li
    className="sc-bdfBwQ CGZeV"
  >
    <span
      className="sc-dlfnbm jAAmbI"
    >
      2
       
    </span>
    <span
      className="sc-gsTCUz gKLpmB"
    >
      test task
      <span
        className="sc-jSgupP cwbpyv"
      >
        Done
      </span>
    </span>
    <button
      className="sc-hKgILt bzvHwI"
      onClick={[Function]}
      size="40"
    >
      Undone
    </button>
    <button
      className="sc-eCssSg cnoREb"
      onClick={[Function]}
      size="40"
    >
      Delete
    </button>
  </li>
</div>
`;

exports[`AddTask Component should render with given state from Redux store 2`] = `
<div>
  <li
    className="sc-bdfBwQ CGZeV"
  >
    <span
      className="sc-dlfnbm jAAmbI"
    >
      2
       
    </span>
    <span
      className="sc-gsTCUz byAzUf"
    >
      test task
      
    </span>
    <button
      className="sc-hKgILt fCEWoY"
      onClick={[Function]}
      size="40"
    >
      Done
    </button>
    <button
      className="sc-eCssSg cnoREb"
      onClick={[Function]}
      size="40"
    >
      Delete
    </button>
  </li>
</div>
`;
